manifests/resource/upstream/member.pp

```diff
@@ -77,7 +77,10 @@
 
   $_server = ($server =~ Pattern[/^unix:\/([^\/\0]+\/*)*$/]) ? {
```

Comment:
How about using a selector on the type?

```puppet
  $_server = $server ? {
    Pattern[/^unix:\/([^\/\0]+\/*)*$/] => $server,
    Stdlib::IP::Address::V6            => "[${server}]:${port}",
    default                            => "${server}:${port}",
  }
```

Reply:
Yes, that's a lot cleaner. `puppet-lint-unquoted_string-check` picks it up because of https://github.com/voxpupuli/puppet-lint-unquoted_string-check/issues/4 though.